src/lib/storage/devices_list.rb

```diff
@@ -0,0 +1,104 @@
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Yast
+  module Storage
+    class DevicesList
+      include Enumerable
+
+      class << self
+        attr_accessor :device_class
+        attr_accessor :default_delegate
+
+        def list_of(klass)
+          self.device_class = klass
+        end
+
+        def by_default_delegate_to(list)
+          self.default_delegate = list
+        end
+      end
+
+      attr_reader :devicegraph
+
+      def initialize(devicegraph, list: nil)
+        @devicegraph = devicegraph
+        @list = list || full_list
```

Comment:
It needs `@devicegraph` to be already assigned, it cannot be evaluated in advance.